code/game/objects/effects/decals/Cleanable/tar.dm

```diff
@@ -0,0 +1,42 @@
+/obj/effect/decal/cleanable/tar
+	gender = PLURAL
+	name = "tar"
+	desc = "A sticky substance."
+	icon_state = "tar2"
+	/// The turf that the tar is sitting on
+	var/turf/simulated/target
+
+/obj/effect/decal/cleanable/tar/Initialize(mapload)
+	..()
+	if(issimulatedturf(loc)) // Ensure the location is a simulated turf
+		target = src.loc
+		target.slowdown += 10 // Apply the slowdown effect to the turf
+		if(prob(50))
+			icon_state = "tar3"
+
+/obj/effect/decal/cleanable/tar/Destroy()
+	if(target)
+		target.slowdown -= 10
+		return ..()
+
+/obj/effect/decal/cleanable/tar/Moved(atom/OldLoc, Dir, Forced)
+	. = ..()
+	target.slowdown -= 10
+	target = src.loc
+	target.slowdown += 10
+	if(!issimulatedturf(target))  // We remove slowdown in Destroy(), so we run this check after adding the slowdown.
```

Comment:
I'm confused. What's the point of removing 10 slowdown and then immediately readding it a line later?

Reply:
We don't want tar on space tiles, right? So if tar ends up in space, we need to qdel it. But the Destroy() proc means that qdeling tar removes 10 slowdown from the turf, which would then make that space tile speedier to walk on.

Reply:
We remove slowdown from the previous turf, then add slowdown to the turf we just move the tar onto, and then qdel it, removing the slowdown from the turf we moved to.

Reply:
oooh I see now, nvm